### ISSUE TITLE
Settings for point values being worth X money, deduct from a user's wallet

### ISSUE THREAD

**Issue by nabeelio** [nabeelio/phpvms#856]

**Comment by Da-Geek** [comment]
I was thinking about adding "expenses" against Rule Violations that are paid out of the Pilots "Wallet"
so if you break a rule it has a $ value, out of the pilots pocket
as right now there is no "Use" for the Pilot Wallet, except bragging rights

**Comment by Da-Geek** [comment]
While looking at the code I ended up finding this following snippet from the '\app\Services\Finance\PirepFinanceService.php'

`
        $this->financeSvc->debitFromJournal(
            $pirep->airline->journal,
            $pilot_pay,
            $pirep,
            $memo,
            'Pilot Pay',
            'pilot_pay'
        );
`

`
        $this->financeSvc->creditToJournal(
            $pirep->user->journal,
            $pilot_pay,
            $pirep,
            $memo,
            'Pilot Pay',
            'pilot_pay'
        );
`

I think this could be used to Debit a Pilots Wallet and Credit the Airlines, so each Violation would cost the Pilot some Money, 
and Pay the Airline that Money for the Violations or Damage to a Plane for Hard Landing and so on...

**Comment by nabeelio** [comment]
Yeah, I was also thinking it should come out from the pilot's wallet and go to the airline.

**Comment by Da-Geek** [comment]
Since violations only arrear from Acars PIREPs, this additional code function could be added to the existing PirepFinanceService.php

You/We would only need to add a New "Tick Box" as well as an Input Box in the "pireps" Settings in admin panel, the
Tick Box to enable the feature, and the Input Box to enter the "Cost" of each violation.

**Comment by nabeelio** [comment]
And also modifying and including additional integration tests to cover this case

**Comment by Da-Geek** [comment]
I was going to start looking into this soon(tm)
I think I can get it working without much of an issue with a hard coded Cost for each violation, the thing I would need to learn would be adding the settings to the Admin Settings, that is what I was thinking of trying to find in the code, and see if I can replicate the current code to do as I want it to.

**Comment by aviezab** [comment]
[![Banner](https://i.ytimg.com/vi/0jk5fGH0MYE/maxresdefault.jpg)](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
jaconey, @paulschmeida, @mskalra, @inguelberth, @4adamu, @kranthisai, @bbodien, @crazy-choose, @ltoshea, @trevex, @ncoelho, @Enigma10, @xfo-0, @hkjels, @agdolla, @nidhigadhavi, @51396, @IsaacBell, @geduardcatalindev, @htadashi, @nonamez, @rotkreis, @tommuhm, @jmlucjav, @MasonStone, @plck1, @matteoscurati, @donpdonp, @woojae-jang, @telemakhos, @theodesp, @aakritishroff, @ivanlemeshev, @shuklaravi, @oiovoyo, @sandroci, @titmy, @samholt, @stopyellingatme, @ryan-shaw, @sensarliar, @refugeesus, @yijun-lee, @ecoinomist, @Rookie39, @pushkarnagle, @datagutt, @kznmft, @viktorbezdek, @panayi
